### PR TITLE
Feature v0.10.0 traffic encrypt

### PR DESCRIPTION
**Pull request by 0xdcarns**
- Added elliptical based traffic encryption should
- Added key generation for client/server
- Fixed some permissions and logging

### CODE REVIEW

File: models/structs.go [hunk 1]
```diff
@@ -166,6 +168,20 @@ type ServerUpdateData struct {
 
 // Telemetry - contains UUID of the server and timestamp of last send to posthog
```
Comment (mattkasun) [inline models/structs.go:169]: comment should address the traffic elements

File: logic/nodes.go [hunk 2]
```diff
@@ -58,6 +58,21 @@ func GetSortedNetworkServerNodes(network string) ([]models.Node, error) {
 	return nodes, nil
 }
 
+// GetServerNodes - gets the server nodes of a network
+func GetServerNodes(network string) []models.Node {
+	var nodes, err = GetNetworkNodes(network)
+	var serverNodes = make([]models.Node, 0)
```
Comment (mattkasun) [inline logic/nodes.go:64]: switch order of so that err check immediately follows potential error condition

File: database/database.go [hunk 3]
```diff
@@ -36,13 +38,13 @@ const INT_CLIENTS_TABLE_NAME = "intclients"
 // PEERS_TABLE_NAME - peers table
 const PEERS_TABLE_NAME = "peers"
 
-// SERVERCONF_TABLE_NAME
+// SERVERCONF_TABLE_NAME - stores server conf
 const SERVERCONF_TABLE_NAME = "serverconf"
 
-// SERVER_UUID_TABLE_NAME
+// SERVER_UUID_TABLE_NAME - stores
```
Comment (mattkasun) [inline database/database.go:44]: comment incomplete

File: netclient/ncutils/netclientutils.go [hunk 4]
```diff
@@ -532,3 +540,131 @@ func CheckWG() {
 		log.Println("running userspace WireGuard with " + uspace)
 	}
 }
+
```
Comment (mattkasun) [inline netclient/ncutils/netclientutils.go:543]: Is this the right package for these functions.  Given they are called in both netmaker and netclient, would it make more sense to put them in logic or functions?

File: netclient/ncutils/netclientutils.go [hunk 5]
```diff
@@ -532,3 +540,131 @@ func CheckWG() {
...
+func BoxDecrypt(encrypted []byte, senderPublicKey *[32]byte, recipientPrivateKey *[32]byte) ([]byte, error) {
+	var decryptNonce [24]byte
+	copy(decryptNonce[:], encrypted[:24])
+	decrypted, ok := box.Open(nil, encrypted[24:], &decryptNonce, senderPublicKey, recipientPrivateKey)
+	if !ok {
+		return nil, fmt.Errorf("could not decrypt message")
+	}
+	return decrypted, nil
+}
+
```
Comment (mattkasun) [inline netclient/ncutils/netclientutils.go:597]: Delete commented out code

**Review by mattkasun — CHANGES_REQUESTED** [review]
complicated as this has multiple changes, but overall nicely done.

File: netclient/ncutils/netclientutils.go (hunk 4 above)
Comment (0xdcarns) [reply netclient/ncutils/netclientutils.go:543]: can't be avoided in current code base

**Comment by 0xdcarns** [comment]
just rebased on develop

**Pull request merged**